src/main/java/com/zendesk/maxwell/MaxwellConfig.java

```diff
@@ -487,10 +490,12 @@ private void parse(String [] argv) {
 		MaxwellOptionParser parser = buildOptionParser();
 		OptionSet options = parser.parse(argv);
 
-		Properties properties;
+		final Properties properties;
 
 		if (options.has("config")) {
 			properties = parseFile((String) options.valueOf("config"), true);
+		} else if (options.has("env_config")) {
+			properties = readPropertiesEnv((String) options.valueOf("env_config"));
```

Comment:
thinking maybe we should also parse the config file?  maybe parse the config file first and then override with the env?  something like that.

Reply:
I disagree, making a smaller number of ways to configure makes it easier to reason about. In this case, the user chooses either `config` or `env_config` for configuration and it throws an error if they attempt to choose both. This model is easier for a user trying to configure it in multiple places and needing to understand which configuration is higher in the hierarchy.

Reply:
sure, possibly harder to reason about.  but this is how the other configuration methods work; if you're passing in command line flags we still parse `config.properties`.  If you use `env_config_prefix`, we grab what's in the environment and still parse `config.properties`.  None of the other methods are exclusive so this one shouldn't be either.

Reply:
I don't entirely agree with the reasoning, but I respect your opinion on project direction as the project owner. Changed to make it additive on top of config.properties.

Ordering of overrides is now `cmd line options > env variable prefix > env config > config file`

Reply:
thanks for that.